src/Bundle/CoreBundle/Form/StateType.php

```diff
@@ -0,0 +1,293 @@
+            $view->vars['current_state'] .= $this->settings['places'][$view->vars['value']['state']]['label'];
+        }
+
+        parent::buildView($view, $form, $options);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        parent::configureOptions($resolver);
+
+        $resolver->setDefaults([
+            'label_prefix' => 'state',
+            'settings' => [],
+            'compound' => true
+        ]);
+
+    }
+
+    /**
+     * Check if can transist from current state
+     *
+     * @param string $current_state
+     * @param string $transition_to
+     *
+     * @return bool
+     */
+    public function canTransist(string $current_state, string $transition_to) : bool
+    {
+        // transitions not set
+        if (!isset($this->settings['transitions'][$transition_to]))
+        {
+            return FALSE;
+        }
+
+        $transition = $this->settings['transitions'][$transition_to];
+
+        if (in_array($current_state, $transition['from']))
+        {
+            return TRUE;
+        }
+
+        return FALSE;
+    }
+
+     /**
+     * @param array $transitions
+     *
+     * @return array
+     */
+    private function getChoices(array $transitions) : array
+    {
+        $choices = [];
+
+        foreach ($transitions as $transition_key => $transition) {
+            $choices[$transition['label']] = $transition_key;
+        }
+
+        return $choices;
+    }
+
+    /**
+     * @param string $transition_key
+     *
+     * @return string
+     */
+    private function getPlaceCategory(string $transition_key) : string
+    {
+        $place = $this->settings['transitions'][$transition_key]['to'];
+
+        if (isset($this->settings['places'][$place]['category'])) {
+            return $this->settings['places'][$place]['category'];
+        }
+
+        return "";
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function transform($value)
+    {
+        $new_values = [];
+        
+        $new_values['state'] = $value;
+        
+        // transition can be alway null
+        $new_values['transition'] = null;
+
+        // if no state set (new content), take initial place
+        if (empty($value))
+        {
+            $new_values['state'] = $this->settings['initial_place'];
+        }
+
+        return $new_values;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function reverseTransform($value)
+    {
+        $new_value = $value['state'];
+
+        // if transition is given, set value only if transition is posssible
+        if (isset($value['transition'])
+            && $value['transition'])
+        {
+            $transition_to = $this->settings['transitions'][$value['transition']]['to'];
+
+            if ($this->canTransist($value['state'], $value['transition']))
+            {
+                $new_value = $transition_to;
+            }
+        }
+
+        return $new_value;

```

Comment:
Why do we need the transition value here, does not return `['state' => $value]` would be all we need?